engine/client/socket.go

```diff
@@ -43,7 +43,15 @@ func (p SocketProvider) CheckAgent(ctx context.Context, req *sshforward.CheckAge
 	}
 	switch u.Scheme {
 	case "unix":
-		path := u.Path
+		var path string
+		if u.RawPath != "" {
+			path, err = url.PathUnescape(u.RawPath)
```

Comment:
I don't get understand this :thinking: 

Why is this behavior different from getting `u.Path`? Shouldn't it be the same? From `url.go`'s `setPath`, called during `url.Parse`:

```
func (u *URL) setPath(p string) error {
	path, err := unescape(p, encodePath)
	if err != nil {
		return err
	}
	u.Path = path
```

Reply:
Mmmh, you are right, sorry 🙏 I extended my tests and I realized that I was basing my implementation on some issue a user had, that led to some encoding problems: https://github.com/dagger/dagger/issues/8401.

However, I can't repro, and following this comment, you're basically right.

As we moved from relative path to absolute path, this does not seem necessary anymore: just path traversal are trimmed in current form "../..". It was problematic in the relative path implementation but the Clean is correct in this case